msal4j-sdk/src/main/java/com/microsoft/aad/msal4j/ManagedIdentityErrorResponse.java

```diff
@@ -15,9 +15,26 @@ public class ManagedIdentityErrorResponse {
     @JsonProperty("correlationId")
     private String correlationId;
 
+    //In some MSI scenarios such as Cloud Shell, the actual error info is in a JSON within the main JSON. To parse that second
+    // JSON layer, we need to first pass it into a subclass, parse it using the usual @JsonProperty annotation, and then retrieve the values.
+    @JsonProperty("error")
+    private void parseErrorField(ErrorField errorResponse) {
+        this.error = errorResponse.code;
+        this.message = errorResponse.message;
+    }
+
     @JsonProperty("error")
     private String error;
 
     @JsonProperty("error_description")
     private String errorDescription;
+
+    @Getter
+    private static class ErrorField {
```

Comment:
Is it possible to write a unit test for this?

Reply:
In the latest commit I expanded one of the unit tests to make sure both the new parsing and old parsing are both covered: the `managedIdentityTest_WrongScopes` test will now conditionally check what source type is used, and if cloud shell the HTTP mocker will return a cloud shell style error response.